apidef/api_definitions.go

```diff
@@ -815,8 +817,8 @@ type UpstreamOAuth struct {
 // PasswordAuthentication holds the configuration for upstream OAuth2 password authentication flow.
 type PasswordAuthentication struct {
 	ClientAuthData
-	// Enabled activates upstream OAuth2 password authentication.
-	Enabled bool `bson:"enabled" json:"enabled"`
+	// Header holds the configuration for the custom header to be used for OAuth authentication.
+	Header HeaderData `bson:"header" json:"header"`
```

Comment:
This could be of type [AuthSource](https://github.com/TykTechnologies/tyk/blob/cea1df441fbf763fb3fd81e2f5a0a38ccadecae6/apidef/oas/authentication.go#L316).